powerauth-restful-security-javaee/src/main/java/io/getlime/security/powerauth/rest/api/jaxrs/controller/v3/TokenController.java

```diff
@@ -0,0 +1,182 @@
+import io.getlime.security.powerauth.rest.api.base.authentication.PowerAuthApiAuthentication;
+import io.getlime.security.powerauth.rest.api.base.encryption.PowerAuthEciesEncryption;
+import io.getlime.security.powerauth.rest.api.base.exception.PowerAuthAuthenticationException;
+import io.getlime.security.powerauth.rest.api.jaxrs.converter.v3.SignatureTypeConverter;
+import io.getlime.security.powerauth.rest.api.jaxrs.provider.PowerAuthAuthenticationProvider;
+import io.getlime.security.powerauth.rest.api.jaxrs.provider.PowerAuthEncryptionProvider;
+import io.getlime.security.powerauth.rest.api.model.request.v3.TokenCreateRequest;
+import io.getlime.security.powerauth.rest.api.model.request.v3.TokenRemoveRequest;
+import io.getlime.security.powerauth.rest.api.model.response.v3.TokenCreateResponse;
+import io.getlime.security.powerauth.rest.api.model.response.v3.TokenRemoveResponse;
+import io.getlime.security.powerauth.soap.axis.client.PowerAuthServiceClient;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import java.util.Arrays;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Controller responsible for publishing services related to simple token-based authentication.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Path("pa/v3/token")
+@Produces(MediaType.APPLICATION_JSON)
+public class TokenController {
+
+    @Inject
+    private PowerAuthServiceClient powerAuthClient;
+
+    @Inject
+    private PowerAuthAuthenticationProvider authenticationProvider;
+
+    @Inject
+    private PowerAuthEncryptionProvider encryptionProvider;
+
+    @Context
+    private HttpServletRequest httpServletRequest;
+
+    @POST
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("create")
+    public ObjectResponse<TokenCreateResponse> createToken(ObjectRequest<TokenCreateRequest> request,
+                                                           @HeaderParam(PowerAuthTokenHttpHeader.HEADER_NAME) String tokenHeader,
+                                                           @HeaderParam(PowerAuthEncryptionHttpHeader.HEADER_NAME) String encryptionHeader) throws PowerAuthAuthenticationException {
+
+        try {
+
+            PowerAuthApiAuthentication authentication = authenticationProvider.validateToken(tokenHeader, Arrays.asList(
+                    PowerAuthSignatureTypes.POSSESSION,
+                    PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE,
+                    PowerAuthSignatureTypes.POSSESSION_BIOMETRY,
+                    PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE_BIOMETRY
+            ));
+
+            PowerAuthEciesEncryption encryption = encryptionProvider.validateEciesEncryption(encryptionHeader);
+
+            if (authentication != null && authentication.getActivationId() != null
```

Comment:
`authentication.getActivationId().equals(encryption.getActivationId())`

This is the code we should try to avoid - or else everyone will have to write this code whenever using E2EE with signatures. Why do we have the E2EE header for Layer2 encryption anyway? Alternatively, we can just ignore the header.

Reply:
As agreed we will not send encryption HTTP header in case the request signature is verified in activation scope, so we can simply remove these checks and use `activationId` and `applicationKey` from HTTP signature.